modules/process/manager.go

```diff
@@ -141,6 +156,9 @@ func (pm *Manager) ExecDirEnvStdIn(timeout time.Duration, dir, desc string, env
 func (pm *Manager) Kill(pid int64) error {
 	if proc, exists := pm.Processes[pid]; exists {
 		pm.mutex.Lock()
+		if proc.Cancel != nil {
+			proc.Cancel()
```

Comment:
`pm.mutex` is locked; what if `proc.Cancel()` is lengthy?

Reply:
Tbh that call is not used in the whole of Gitea and is actually incorrect anyway.

The process manager needs to lock before it looks up processes.

Reply:
> 
> 
> The process manager needs to lock before it looks up processes.

Agree. To keep this PR on-topic, what about adding a comment at the top of the function with a notice/warning?

Reply:
I may aswell fix it or destroy it. 

Hmm I could destroy it - cancel is a nicer thing than kill. We can always restore it if we decide we need kill in the end.

Reply:
> 
> Hmm I could destroy it - cancel is a nicer thing than kill. We can always restore it if we decide we need kill in the end.

I think that'll do. 👍

Reply:
I've removed it and the now unused reference to cmd... 

Now, comes the fun - Process can now be a general cancellable context watcher.